File: RISCV_ASM/examples/factorial/factorial.c
#include <stdio.h>
int factorial(int n) {
    if (n == 0) {
        return 1;
    }
    return n * factorial(n-1); 
}

int main() {
    int n;
    if (scanf("%d", &n) != 1) {
        return -1;	
    }
    printf("%d\n", factorial(n));
    return 0;
}

File: CSharpStrange/bin/factorial.cs
public class Program
{
    public int Factorial(int n)
    {
        if (n == 0)
        {
            return 1;
        }
        else
        {
            return n * Factorial(n - 1);
        }
    }

    public static void Main()
    {

    }
}